sbi/inference/snpe/snpe_c.py

```diff
@@ -161,7 +161,8 @@ def train(
             # last proposal.
             proposal = self._proposal_roundwise[-1]
             self.use_non_atomic_loss = (
-                isinstance(proposal.posterior_estimator._distribution, mdn)
+                isinstance(proposal, DirectPosterior)
```

Comment:
how does this check change the behaviour for `MCMCPosterior` or `MultivariateNormal`?

Reply:
previously, the first check was `isinstance(proposal.posterior_estimator._distribution, mdn)`. This raised an error if, e.g. `proposal` was an `MCMCPosterior`.

Now, the first check is `isinstance(proposal, DirectPosterior)` which will be `False` if `proposal is, e.g., an `MCMCPosterior`. Because everything below is an `and` case, these lines will never be executed.

Reply:
See https://stackoverflow.com/questions/13960657/does-python-evaluate-ifs-conditions-lazily

Reply:
the Python behaviour is clear to me, I was wondering about the `sbi` semantic. But true, we want the line to be `False` for `MCMCPosterior` and other, instead of throwing an error. so all good!